.github/workflows/notify-merge.js

```diff
@@ -0,0 +1,123 @@
+const fs = require('fs');
+const { slackNotification } = require('./helpers.js');
+
+const SLACK = {
+    merge: ({ html_url, number, title, prefix = '' }) =>
+        `:merged: PR merged to stage: ${prefix} <${html_url}|#${number}: ${title}>.`,
+};
+
+const mergeRegex = /Merge pull request #(\d+)/i;
+function getRepoInfo() {
+    const repoFull = process.env.GITHUB_REPOSITORY;
+    if (!repoFull) {
+        throw new Error("GITHUB_REPOSITORY is not defined in the environment.");
+    }
+    const [owner, repo] = repoFull.split('/');
+    return { owner, repo };
+}
+
+async function getPR(prNumber, owner, repo, token) {
+    const url = `https://api.github.com/repos/${owner}/${repo}/pulls/${prNumber}`;
+    const response = await fetch(url, {
+        headers: {
+            Accept: 'application/vnd.github.v3+json',
+            Authorization: `token ${token}`,
+        },
+    });
+    if (!response.ok) {
+        throw new Error(`Failed to fetch PR ${prNumber}: ${response.status}`);
+    }
+    return await response.json();
+}
+
+async function getAssociatedPR(commitSha, owner, repo, token) {
+    const url = `https://api.github.com/repos/${owner}/${repo}/commits/${commitSha}/pulls`;
+    const response = await fetch(url, {
+        headers: {
+            Accept: 'application/vnd.github.groot-preview+json',
+            Authorization: `token ${token}`,
+        },
+    });
+    if (!response.ok) {
+        throw new Error(`Failed to fetch associated PR for commit ${commitSha}: ${response.status}`);
+    }
+    const prs = await response.json();
+    return prs.length > 0 ? prs[0] : null;
+}
+
+async function main() {
+    let prNumbers = new Set();
+
+    // Read the event payload from GitHub Actions.
+    const eventPath = process.env.GITHUB_EVENT_PATH;
+    if (eventPath) {
+        try {
+            const eventData = JSON.parse(fs.readFileSync(eventPath, 'utf8'));
+            const commits = eventData.commits || [];
+            commits.forEach((commit) => {
+                const message = commit.message || '';
+                const match = mergeRegex.exec(message);
+                if (match && match[1]) {
+                    prNumbers.add(match[1]);
+                }
+            });
+        } catch (error) {
+            console.error('Error reading event payload:', error);
+        }
+    }
```

Comment:
I don't get the sense of this, don't you already get the PR number in the payload of the action when a PR is merged, since this is only triggered on merge?